cmake/Platform/Linux/Packaging/o3de.assetprocessor.in

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+pushd @CPACK_PACKAGING_INSTALL_PREFIX@
+bin/Linux/profile/Default/AssetProcessor $@
+popd
```

Comment:
Instead of changing the user working directory when running the AssetProcessor, Editor and ProjectManager, does it work with prepending the `$CPACK_PACKAGING_INSTALL_PREFIX` value to the command that executes.
Such as
```suggestion
@CPACK_PACKAGING_INSTALL_PREFIX@/bin/Linux/profile/Default/AssetProcessor "$@"
```

Reply:
The change you mentioned seems to work, I just went with the pushd way out of caution since some a lot of the scripts seem to need to be run from the engine root. It may not be necessary though, we can change it to your version and if any problems pop up we can switch back to the more verbose version.

Reply:
The C++ Applications could be launched with a relative path to a user's project.
Changing directories would break that.

For example if I had a project in `/home/lumberyard-employee-dm/MyProject` and my current directory was the same.
Then I could launch the Editor as follows. to have it load the Gems for MyProject
```bash
[lumberyard-employee-dm@arch MyProject]$ Editor --project-path .
```

Reply:
I don't think you even need to pushd/popd since the executables are being ran in a bash script. You can probably just use `cd` instead

Reply:
I don't think you even need to pushd/popd since the executables are being ran in a bash script. You can probably just use `cd` instead

```suggestion
cd @CPACK_PACKAGING_INSTALL_PREFIX@ && bin/Linux/profile/Default/AssetProcessor $@
```

Reply:
Actually this script doesn't need to be a bash script.
It could be a regular `sh` bourne shell script.
There is no bash specific functionality and it would make it a bit more portable if used with another shell such as `zsh`.

Reply:
pushd/popd is not supported in `sh`, so if its changed to `cd` then `sh` should be good enough.

Reply:
A change of directory shouldn't be used at all.
Changing directory messes up the parameters the user supplied to the application which are relative paths.

So we shouldn't change directory in the first place.

Reply:
In that case, shouldn't this be a symlink instead of a script redirect?

Reply:
Symlink sounds like the better choice. In addition, I suggest using `CMAKE_RUNTIME_OUTPUT_DIRECTORY` so we don't pin this to the `profile` target path, though the variable may need to be copied to CPack via `Packaging.cmake`

Reply:
The only problem with symlinks is that we can't do things like specify environment variables, add checks or other extra logic, etc. That may not matter for now though, so we can switch over if having entire scripts for each executable is overkill.

Reply:
So `/o3de/build/linux/bin` is certainly not the correct path.
It would be something like `bin/Linux/profile/Default`

Reply:
Do you know if there happens to be any cmake variables we can make use of to get the "bin/Linux/profile/Default" path string, without hardcoding the platform and configuration like I'm currently doing?

Reply:
Well we would probably have add the relative path from the install root to the RUNTIME_OUTPUT_DIRECTORY as part of the generated [CPackComponents.cmake](https://github.com/o3de/o3de/blob/development/cmake/Packaging.cmake#L186-L197) file.

If we take the logic from the [Install_common.cmake](https://github.com/o3de/o3de/blob/development/cmake/Platform/Common/Install_common.cmake#L50-L57) which creates a relative directory to the RUNTIME/LIBRARY/ARCHIVE_OUTPUT_DIRECTORIES, we could use that to inject a relative path into the CPack
```ccmake
cmake_path(RELATIVE_PATH CMAKE_RUNTIME_OUTPUT_DIRECTORY BASE_DIRECTORY ${CMAKE_BINARY_DIR} OUTPUT_VARIABLE runtime_output_directory)
cmake_path(RELATIVE_PATH CMAKE_LIBRARY_OUTPUT_DIRECTORY BASE_DIRECTORY ${CMAKE_BINARY_DIR} OUTPUT_VARIABLE library_output_directory)
# Get the output folders, archive is always the same, but runtime/library can be in subfolders defined per target
cmake_path(RELATIVE_PATH CMAKE_ARCHIVE_OUTPUT_DIRECTORY BASE_DIRECTORY ${CMAKE_BINARY_DIR} OUTPUT_VARIABLE archive_output_directory)


cmake_path(APPEND archive_output_directory "${PAL_PLATFORM_NAME}/$<CONFIG>/${LY_BUILD_PERMUTATION}")
cmake_path(APPEND library_output_directory "${PAL_PLATFORM_NAME}/$<CONFIG>/${LY_BUILD_PERMUTATION}")
cmake_path(APPEND runtime_output_directory "${PAL_PLATFORM_NAME}/$<CONFIG>/${LY_BUILD_PERMUTATION}")
```